### PR TITLE
feat: logger

### PR DESCRIPTION
**Pull request by lemaitre-aneo**

### CODE REVIEW

File: packages/cpp/ArmoniK.Api.Common/source/logger/formatter.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    // Get current time
+    auto time = std::chrono::system_clock::now();
+    auto ns = std::chrono::duration_cast<std::chrono::nanoseconds>(time.time_since_epoch()).count() % 1'000'000'000;
+
+    // Format message with timestamp and level
+    fmt::format_to(out, FMT_COMPILE(R"({{"@t": "{:%FT%T}.{:09}Z", "@mt": {:?}, "@l": {:?})"), fmt::gmtime(time), ns,
+                   message, level_name(level));
+
+    // Add contexts to the formatted
```
Comment (dbrasseur-aneo) [inline packages/cpp/ArmoniK.Api.Common/source/logger/formatter.cpp:37]:
```suggestion
    // Add contexts to the formatted string
```

File: packages/cpp/ArmoniK.Api.Common/source/logger/formatter.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    for (auto context : {&global_context, &local_context, &message_context}) {
+      for (auto &[key, val] : *context) {
+        fmt::format_to(out, FMT_COMPILE(", {:?}: {:?}"), key, val);
```
Comment (dbrasseur-aneo) [inline packages/cpp/ArmoniK.Api.Common/source/logger/formatter.cpp:40]:
```suggestion
        fmt::format_to(out, FMT_COMPILE(R", "{:?}": "{:?}""), key, val);
```

File: packages/cpp/ArmoniK.Api.Worker/header/utils/WorkerServer.h [hunk 3]
```diff
@@ -41,9 +42,14 @@ class WorkerServer {
    * @brief Constructor for the WorkerServer class.
    * @param configuration A shared pointer to the Configuration object.
    */
-  explicit WorkerServer(const Common::utils::Configuration &configuration) {
-    logger.enrich([](Common::serilog::serilog_context &ctx) { ctx.add("threadId", std::this_thread::get_id()); });
-    logger.add_property("container", "ArmoniK.Worker");
+  explicit WorkerServer(const Common::utils::Configuration &configuration)
+      : logger(Common::logger::writer_console(), Common::logger::formatter_clef()) {
+    logger.local_context_generator_add("threadId", []() {
+      std::stringstream ss;
```
Comment (dbrasseur-aneo) [inline packages/cpp/ArmoniK.Api.Worker/header/utils/WorkerServer.h:48]: Could you be missing #include <sstream> ?

Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Worker/header/utils/WorkerServer.h:48]: As you pointed out offline, it was `#include <thread>`

File: packages/cpp/ArmoniK.Api.Common/source/logger/formatter.cpp (hunk 2 above)
Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Common/source/logger/formatter.cpp:39]: If I recall correctly, `{:?}` includes quotes

File: packages/cpp/ArmoniK.Api.Common/header/logger/local_logger.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  /**
+   * @brief Remove an entry from the context.
+   * @param key Name of the entry to remove.
+   */
+  void context_remove(const std::string &key);
+
+public:
+  /**
+   * @copydoc ILogger::log()
+   */
```
[the remark this answers is not in the document]
Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Common/header/logger/local_logger.h:74]: I prefer to be explicit here. I think it makes it simpler to read the code.

File: packages/cpp/ArmoniK.Api.Common/header/logger/context.h [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+namespace API_COMMON_NAMESPACE::logger {
+/**
+ * @class Context
+ * @brief Logger context.
+ */
+class Context : public std::map<std::string, std::string> {
+public:
+  using std::map<std::string, std::string>::map;
+  using std::map<std::string, std::string>::operator=;
+};
```
[the remark this answers is not in the document]
Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Common/header/logger/context.h:19]: Could be, but this makes it a new and different type.

**Review by dbrasseur-aneo — APPROVED** [review]

**Pull request merged**